main.go

```diff
@@ -31,6 +31,398 @@ const bwSuccessPrefix = "✓"
 const bwFailPrefix = "𐄂"
 const bwErrorPrefix = "!"
 
+/**
+ * Call function and, when it is possible display spinner with some progress message
+ */
+func showProgress(progressMessage string, isColorful bool, function func() error) error {
+	var s *spinner.Spinner
+	if isColorful {
+		s = spinner.New(spinner.CharSets[9], 100*time.Millisecond)
+		s.Suffix = progressMessage
+		s.Start()
+	}
+
+	err := function()
+
+	if isColorful {
+		s.Stop()
+	}
+
+	return err
```

Comment:
`defer` can be used to streamline this code a bit.

```suggestion
	defer func() {
	    if isColorful {
	        s.Stop()
	    }
	}
	return function()
```

Reply:
Good idea, and I think that we can streamline it little bit more.